File: 014 - Longest Common Prefix (Easy)/Python/solution.py
class Solution:
    def longestCommonPrefix(self, strs):
        if not strs: return ""                                      # If there are no strings, return an empty string.
        
        minString = min(strs)                                       # Find the minimum string
        maxString = max(strs)                                       # Find the maximum string
        
        for index, char in enumerate(minString):                    # For each character in the min string
            if char != maxString[index]: return minString[:index]   # Return a slice if it doesn't match the max string character
            
        return minString                                            # Return minString
        

File: 014 - Longest Common Prefix (Easy)/C++/solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.size() < 1) return "";                                                     // If strs is empty, return an empty string                                           
    
        string prefix = strs[0];                                                            // Initialize the prefix to the first string
    
        for(int i = 1; i < strs.size(); i++){                                               // For each string
            int j = 0;                                                                      // Create a new iterator and initialize it to 0
            
            while(j < min(prefix.size(), strs[i].size()) && strs[i][j] == prefix[j]) j++;   // Increment until strings are no longer equal
            
            prefix = strs[i].substr(0, j);                                                  // Slive the prefix down based on the iterator
        }
                  
        return prefix;                                                                      // Return prefix
    }
};